compiler/rustc_ast/src/ast.rs

```diff
@@ -1282,6 +1282,7 @@ pub struct Closure {
     pub body: P<Expr>,
     /// The span of the argument block `|...|`.
     pub fn_decl_span: Span,
+    pub cl_pipe_start: BytePos,
```

Comment:
Could you add a comment?
Should we just have a span for `move || -> ty` and one for just `||`?
Like `fn_arg_span` you made for HIR.

Reply:
I wrote this before #101562 to try to minimize the size of `ast::Expr`. Now that `Closure` is boxed I agree changing this to a span of `|...|` will be better.